cmd/erasure-sets.go

```diff
@@ -217,6 +217,10 @@ func (s *erasureSets) connectDisks() {
 				continue
 			}
 		}
+		if cdisk != nil {
+			// Close previous offline disk.
+			cdisk.Close()
```

Comment:
It doesn't look like this is an offline disk; diskMap already contains only onlineDisks(); aren't we closing a live offline disk here ?

Reply:
No we are closing an offline one @vadmeste since the DiskLoc() returns `-1`

Reply:
well, DiskLoc() will not return -1 for an offline disk, we are not setting set to -1 when a disk goes offline. I am not able to find that anywhere.